### PR TITLE
Update Python versions in the CI workflow.

### PR DESCRIPTION
**Pull request by ezio-melotti**
Same as:
* https://github.com/python/miss-islington/pull/590
* https://github.com/python/bedevere/pull/520
* https://github.com/python/blurb_it/pull/316

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -6,19 +6,14 @@ jobs:
...
     strategy:
       matrix:
-        python:
-          - '3.7'
-          - '3.8'
-          - '3.9'
-          - '3.10'
-          - '3.11'
-    name: ${{ matrix.python }}
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12-dev"]
```
Comment (AlexWaygood) [inline .github/workflows/tests.yml:9]: Was 3.7 dropped deliberately?

```suggestion
        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12-dev"]
```

Comment (ezio-melotti) [reply .github/workflows/tests.yml]: Good catch, that was accidental :)

**Review by AlexWaygood — APPROVED** [review]
LGTM

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**